project/CelebornBuild.scala

```diff
@@ -205,6 +205,18 @@ object CelebornCommonSettings {
     Dependencies.junit % "test",
     // https://www.scala-sbt.org/1.x/docs/Testing.html
     Dependencies.junitInterface % "test")
+
+  def licenseFileOnlyProjectMergeStrategy(log: Logger) =
```

Comment:
```bash
[info] assembly - excluding file: Jar name = commons-lang3-3.12.0.jar, jar org = org.apache.commons, entry target = META-INF/LICENSE.txt
[info] assembly - excluding file: Jar name = commons-lang3-3.12.0.jar, jar org = org.apache.commons, entry target = META-INF/NOTICE.txt
[info] assembly - including file: META-INF/NOTICE
[info] assembly - including file: META-INF/LICENSE
```

Reply:
I think the built-in strategy `MergeStrategy.preferProject` can achieve this goal

[merge strategy](https://github.com/sbt/sbt-assembly#merge-strategy)

> `MergeStrategy.preferProject` will choose the first project file over library files if present. Otherwise, it works like `MergeStrategy.first`

Reply:
Maybe not? If our project does not have the same file

https://github.com/sbt/sbt-assembly/blob/e4d3b362997a32da21d97c051c9d92ecc5cdb9c2/src/main/scala/sbtassembly/MergeStrategy.scala#L82-L88

sbtassembly.MergeStrategy#preferProject
```scala
  /**
   * Picks the dependency that is the first project/internal dependency, otherwise picks the first library/external dependency
   */
  val preferProject: MergeStrategy = MergeStrategy("PreferProject") { conflicts =>
    val entry = conflicts.find(_.isProjectDependency).getOrElse(conflicts.head)
    Right(Vector(JarEntry(entry.target, entry.stream)))
  }
```

Reply:
is it feasible to combine using `MergeStrategy.preferProject` and `MergeStrategy.discard`?

```
case "LICENSE" | "NOTICE" => MergeStrategy.preferProject
case m if Assembly.isLicenseFile(m) => MergeStrategy.discard
```

Reply:
```scala
          case PathList(ps@_*) if ps.last == "LICENSE" || ps.last == "NOTICE" => MergeStrategy.preferProject
          case PathList(ps@_*) if Assembly.isLicenseFile(ps.last) => MergeStrategy.discard
```

<img width="783" alt="image" src="https://github.com/apache/incubator-celeborn/assets/3898450/01d6f68c-d9ad-44c2-a16f-47a0f756f267">